fix udp src & dst targets

archercreat opened the pull request:
Hello. The UDP hook had remote and local targets switched. This pr fixes the issue plus adds real local ip and local port. Unfortunately the offsets are undocumented so I hardcoded whem.

tklengyel commented on src/plugins/socketmon/socketmon.cpp:
```diff
@@ -461,6 +463,30 @@ static event_response_t udp_send_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info
+    if (s->build.buildnumber == win_7_sp1_ver)
+        poff = 0x80;
+    else if (s->build.buildnumber == win_10_1803_ver)
+        poff = 0x78;
```
Can we collect these offsets in an array at least if they can't be found in a json profile?

archercreat replied:
yes, but I found another bug in udp so I'll add it to this pr later.

tklengyel commented:
@drakvuf-jenkins Test this please

drakvuf-jenkins commented:
Can one of the admins verify this patch?

archercreat commented:
There were quite a few bugs in socketmon plugin. 
- Added support for the second type of UDP connection. I'm not sure what this type called, but socketmon did not track simple `ping google.com` UDP packets...
- Added support for x86 and some major Windows versions;
- Converted Ipv6 address back to Ipv4 for dual-socket connections. Basically Ipv4 address is passed to tcpip driver as Ipv6 address and converted back before actual transmission. Because of this, you will see Ipv4 address in WireShark but Ipv6 address in socketmon. More info here: https://learn.microsoft.com/en-us/windows/win32/winsock/dual-stack-sockets

tklengyel commented:
@drakvuf-jenkins Test this please
